appstore/developer_portal_api.py

```diff
@@ -0,0 +1,558 @@
+                app_uuid=appinfo['uuid'],
+                asset_collections={x: AssetCollection(
+                    platform=x,
+                    description=params['description'],
+                    screenshots=[upload_asset(s, s.content_type) for s in screenshots[x]],
+                    headers=[header_asset] if header_asset else [],
+                    banner=None
+                ) for x in screenshots},
+                category_id=category_map[params['category']],
+                companions={}, # companions not supported yet
+                created_at=datetime.datetime.utcnow(),
+                developer=developer,
+                hearts=0,
+                releases=[],
+                icon_large=upload_asset(request.files['large_icon'], request.files["large_icon"].content_type),
+                icon_small=upload_asset(request.files['small_icon'], request.files["small_icon"].content_type) if 'small_icon' in params else '',
+                source=params['source'] if 'source' in params else "",
+                title=params['title'],
+                type=params['type'],
+                timeline_enabled=False,
+                website=params['website'] if 'source' in params else "",
+            )
+            db.session.add(app_obj)
+            print(f"Created app {app_obj.id}")
+
+            release = release_from_pbw(app_obj, pbw_file,
+                                       release_notes=params['release_notes'],
+                                       published_date=datetime.datetime.utcnow(),
+                                       version=appinfo['versionLabel'],
+                                       compatibility=appinfo.get('targetPlatforms', ['aplite', 'basalt', 'diorite', 'emery']))
+            print(f"Created release {release.id}")
+            upload_pbw(release, request.files['pbw'])
+            db.session.commit()
+
+            if algolia_index:
+                algolia_index.partial_update_objects([algolia_app(app_obj)], { 'createIfNotExists': True })
+
+            try:
+                announce_new_app(app_obj)
+            except Exception:
+                # We don't want to fail just because Discord is being weird
+                print("Discord is being weird")
+
+            return jsonify(success=True, id=app_obj.id)
+
+        else:
+            return jsonify(error=appError, e=appErrorCode), 400
+
+@devportal_api.route('/app/<app_id>', methods=['POST'])
+def update_app_fields(app_id):
+        req = request.json
+
+        if req is None:
+            return jsonify(error="Invalid POST body. Expected JSON", e="body.invalid"), 400
+
+        allowed_fields_type_map = {
+            "title": str,
+            "description": str,
+            "category": str,
+            "website": str,
+            "source": str,
+            "visible": bool
+        }
+
+        # Check all passed fields are allowed
+        for x in req:
+            if not type(x) == allowed_fields_type_map[x]:
```

Comment:
this will unexpectedly KeyError if a field that is passed in isn't in the type map, right?

Reply:
...not anymore